src/axom/primal/geometry/Polyhedron.hpp

```diff
@@ -503,33 +503,39 @@ class Polyhedron
   }
 
   /*!
-   * \brief Computes the signed volume of the polyhedron.
+   * \brief Computes the moments of the polyhedron. The 0th moment is the
+   *        volume of the polyhedron, the 1st moment is the centroid.
    *
-   * \return The signed volume of the polyhedron
+   * \param [out] volume The volume of the polyhedron (0th moment)
+   * \param [out] centroid The centroid of the polyhedron (1st moment)
    *
    * \note Function is based off moments() in Mike Owen's PolyClipper.
    *
    * \pre polyhedron vertex neighbors are defined, and polyhedron is 3D
    *
    * \sa volume()
+   * \sa centroid()
    */
   AXOM_HOST_DEVICE
-  double signedVolume() const
+  void moments(double& volume, PointType& centroid) const
```

Comment:
Possible optimization:

It seems like this implementation will do extra computation in cases where we only want the volume but not the centroid
(Note that the centroid calculation always needs the volume).

 As an optimization, would it make sense to have an extra boolean flag `should_compute_centroid` (possibly defaulting to true) and guard the few operations that only relate to the centroid?

Then `signedVolume()` could be something like:
```cpp
  double signedVolume() const
  {
    double volume;
    PointType centroid;
    moments(volume, centroid, false);
    return volume;
  }
```

I'm raising this issue b/c we compute the polyhedron volume in our intersection-based shaping, and that call can be quite expensive, so we'd want to avoid making it unnecessarily more expensive.